loki/transformations/build_system/file_write.py

```diff
@@ -72,6 +69,6 @@ def transform_file(self, sourcefile, **kwargs):
         path = Path(item.path)
         suffix = self.suffix if self.suffix else path.suffix
         sourcepath = Path(item.path).with_suffix(f'.{self.mode}{suffix}')
-        if self.builddir is not None:
-            sourcepath = self.builddir/sourcepath.name
+        if (builddir := kwargs.get('build_args', {}).get('builddir', None)) is not None:
```

Comment:
I think we're be able to afford two lines to make this a little more readable :)
```suggestion
        build_args = kwargs.get('build_args')
        if build_args and (builddir := build_args.get('builddir')):
```